src/Aspire.Dashboard/Model/DashboardClient.cs

```diff
@@ -124,6 +148,44 @@ GrpcChannel CreateChannel()
                     LoggerFactory = _loggerFactory,
                     ThrowOperationCanceledOnCancellation = true
                 });
+
+            X509CertificateCollection GetFileCertificate()
+            {
+                var filePath = configuration["ResourceServiceClient:ClientCertificate:FilePath"];
+                var password = configuration["ResourceServiceClient:ClientCertificate:Password"];
+
+                if (filePath is null or [])
+                {
+                    throw new InvalidOperationException("ResourceServiceClient:ClientCertificate:Source is \"File\", but no Certificate:FilePath is configured.");
+                }
+
+                return [new X509Certificate2(filePath, password)];
+            }
+
+            X509CertificateCollection GetKeyStoreCertificate()
+            {
+                var subject = configuration["ResourceServiceClient:ClientCertificate:Subject"];
+
+                if (subject is null or [])
+                {
+                    throw new InvalidOperationException("ResourceServiceClient:ClientCertificate:Source is \"KeyStore\", but no Certificate:FilePath is configured.");
+                }
+
+                using var store = new X509Store(storeName: StoreName.My, storeLocation: StoreLocation.CurrentUser);
+
+                configuration.Bind("ResourceServiceClient:ClientCertificate:KeyStore");
```

Comment:
I'm updating this code to use strongly typed options, and I noticed this line. I don't think it does anything.

I'm guessing you were intending to make store configurable like Kestrel, but didn't finish.

Reply:
My understanding of that line is that it allows config values to apply to the store object. 

The idea is documented in https://github.com/dotnet/aspire/tree/main/src/Aspire.Dashboard#resources

Reply:
This line is trying to bind IConfiguration values to a string: [Bind(IConfiguration, Object)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.bind?view=dotnet-plat-ext-8.0#microsoft-extensions-configuration-configurationbinder-bind(microsoft-extensions-configuration-iconfiguration-system-object))

Reply:
Thanks, yes that's a bug.

Reply:
Fixed in #3139.